### PR TITLE
feat(content): Shocking flag

### PR DESCRIPTION
**Pull request by yay855**
Added in the shocking json flag, changes to electrical weapons, electrified rapier, tonfa recipe adjustment.

## Purpose of change

There was a FLAMING flag which made weapons deal bonus fire damage, but not a SHOCKING flag that made weapons deal bonus electrical damage.

## Describe the solution

Added in the SHOCKING flag which is an electrical equivalent to the FLAMING flag. Adjusted electrical weapons to compensate, giving them transformations instead. Made tonfas autolearn, since they're not _that_ amazing and they're basically just clubs with a prong coming out of them.

## Testing

I've managed to get it down to only one error! If _anyone_ can help me, _please do._

### CODE REVIEW

**Comment by chaosvolt** [comment]
Hmm. Question: did these need to be hardcoded in at all? I see nothing in this so far that can't be replicated using a mundane transform action?

**Comment by yay855** [comment]
> Hmm. Question: did these need to be hardcoded in at all? I see nothing in this so far that can't be replicated using a mundane transform action?

No, no it did not.

**Comment by yay855** [comment]
Thanks for catching that, Chaos.

File: data/json/items/melee/bludgeons.json [hunk 1]
```diff
@@ -762,6 +768,25 @@
...
+    "name": { "str": "powered quarterstaff (on)", "str_pl": "powered quarterstaves (on)" },
+    "description": "This is an ironshod quarterstaff that has a high-voltage stun gun built into the handle.  The stun gun is wired to the metal caps at either end of the staff, allowing you to zap a dangerous opponent with each strike.  It is currently on.",
+    "revert_to": "shock_staff",
+    "use_action": {
+      "type": "transform",
+      "msg": "You press a button and the staff turns off.",
+      "active": true,
+      "need_charges": 1,
+      "need_charges_msg": "The %s's batteries are dead.",
+      "target": "shock_staff"
```
Comment (chaosvolt) [inline data/json/items/melee/bludgeons.json:784]:
```suggestion
    "revert_to": "shock_staff",
    "revert_msg": "The %s's batteries run dry.",
    "use_action": {
      "type": "transform",
      "msg": "You press a button and the staff turns off.",
      "target": "shock_staff"
```

File: data/json/items/melee/bludgeons.json [hunk 2]
```diff
@@ -882,10 +913,11 @@
     "copy-from": "shocktonfa_off",
     "type": "TOOL",
     "name": { "str": "tactical tonfa (on)", "str_pl": "tactical tonfas (on)" },
-    "description": "This is a reinforced plastic tonfa; the core is hollowed out and is filled with capacitors and a high-yield rechargeable storage battery.  When a switch on the handle is pressed, a high-voltage current is transmitted to the two electrodes mounted in the end of the tonfa, and by extension to anyone unfortunate enough to be in contact with them.  It also has a nifty flashlight, which is currently active.",
+    "description": "This is a reinforced plastic tonfa; the core is hollowed out and is filled with capacitors and a battery compartment.  When a switch on the handle is pressed, a high-voltage current is transmitted to the two electrodes mounted in the end of the tonfa, and by extension to anyone unfortunate enough to be in contact with them.  It's currently turned on.",
     "revert_to": "shocktonfa_off",
```
Comment (chaosvolt) [inline data/json/items/melee/bludgeons.json:917]:
```suggestion
    "revert_to": "shocktonfa_off",
    "revert_msg": "The %s's batteries run dry.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 3]
```diff
@@ -1123,6 +1129,18 @@
     ],
     "magazine_well": "500 ml"
   },
+  {
+    "id": "shock_foil_on",
+    "copy-from": "shock_foil",
+    "type": "TOOL",
+    "name": { "str": "electrified foil (on)", "str_pl": "electrified foils (on)" },
+    "description": "The modifications made to this fencing foil may not be sporting, but they might give you an edge in this final competition.  Extra effort has been made to insulate the grip and a high voltage stun gun has been attached to the guard, wired through the electronic scoring circuit to the tip.  It is currently turned on.",
+    "revert_to": "shock_foil",
```
Comment (chaosvolt) [inline data/json/items/melee/swords_and_blades.json:1138]:
```suggestion
    "revert_to": "shock_foil",
    "revert_msg": "The %s's batteries run dry.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 4]
```diff
@@ -1163,6 +1187,18 @@
     ],
     "magazine_well": "500 ml"
   },
+  {
+    "id": "shock_epee_on",
+    "copy-from": "shock_epee",
+    "type": "TOOL",
+    "name": { "str": "electrified epee (on)", "str_pl": "electrified epees (on)" },
+    "description": "The modifications made to this fencing épée may not be sporting, but they might give you an edge in this final competition.  Extra effort has been made to insulate the grip and a high voltage stun gun has been attached to the guard, wired through the electronic scoring circuit to the tip.  It is currently turned on.",
+    "revert_to": "shock_epee",
```
Comment (chaosvolt) [inline data/json/items/melee/swords_and_blades.json:1196]:
```suggestion
    "revert_to": "shock_epee",
    "revert_msg": "The %s's batteries run dry.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 5]
```diff
@@ -1202,6 +1244,18 @@
     ],
     "magazine_well": "500 ml"
   },
+  {
+    "id": "shock_sabre_on",
+    "copy-from": "shock_sabre",
+    "type": "TOOL",
+    "name": { "str": "electrified saber (on)", "str_pl": "electrified sabers (on)" },
+    "description": "The modifications made to this fencing saber may not be sporting, but they might give you an edge in this final competition.  Extra effort has been made to insulate the grip and a high voltage stun gun has been attached to the guard, wired through the electronic scoring circuit to the tip.  It is currently turned on.",
+    "revert_to": "shock_sabre",
```
Comment (chaosvolt) [inline data/json/items/melee/swords_and_blades.json:1253]:
```suggestion
    "revert_to": "shock_sabre",
    "revert_msg": "The %s's batteries run dry.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 6]
```diff
@@ -1291,6 +1345,44 @@
...
+    ],
+    "magazine_well": "500 ml"
+  },
+  {
+    "id": "shock_rapier_on",
+    "copy-from": "shock_rapier_off",
+    "type": "TOOL",
+    "name": { "str": "electrified rapier (on)", "str_pl": "electrified rapiers (on)" },
+    "description": "The modifications made to this rapier may not be sporting, but they might give you an edge in this final competition.  Extra effort has been made to insulate the grip and a high voltage stun gun has been attached to the guard, wired through the electronic scoring circuit to the tip.  It is currently turned on.",
+    "revert_to": "shock_rapier_off",
```
Comment (chaosvolt) [inline data/json/items/melee/swords_and_blades.json:1380]:
```suggestion
    "revert_to": "shock_rapier_off",
    "revert_msg": "The %s's batteries run dry.",
```

**Review by chaosvolt — COMMENTED** [review]
Mostly just one case of a turn-off action still retaining active toggle and other stuff copied from the turn-on action, but also having a separate revert message could be an optional improvement if desired, so it doesn't just claim you flick the switch off automatically when it runs out of charges and reverts.

**Review by chaosvolt — APPROVED** [review]

**Comment by VissValdyr** [comment]
How about a shocking pride flag now?

**Pull request merged**